setup.sh

```diff
@@ -232,27 +232,60 @@ setup_networking() {
 		cidr=$(echo $TINKERBELL_NETWORK | grep -Eo "\/[[:digit:]]+" | grep -v "^$" | tr -d "/")
 		case "$1" in 
 			ubuntu)
-				if [ ! -f /etc/network/interfaces ] ; then
-					echo "$ERR file /etc/network/interfaces not found"
-					exit 1
-				fi
+				v16=16.04
+				version=$(. /etc/os-release && echo "$VERSION_ID")
+				if (( $(echo "$version > $v16" | bc -l) )); then
+					echo "$INFO found Ubuntu $version, using 'netplan' for network configuration"
+
+					# this is required to fix the bug descibed here
+					# https://bugs.launchpad.net/netplan/+bug/1810043
+					apt-get install -y --no-install-recommends netplan.io >> /dev/null 
+					cat >/etc/netplan/${TINKERBELL_NETWORK_INTERFACE}.yaml <<EOF
+---
+network:
+  version: 2
+  renderer: networkd
+  ethernets:
+    $TINKERBELL_NETWORK_INTERFACE:
+      addresses:
+        - $TINKERBELL_HOST_IP/$cidr
+        - $TINKERBELL_NGINX_IP/$cidr
+EOF
```

Comment:
Generating YAML on the fly with bash feels a bit scary. In particular, magic values can turn in to other things like "no" being a bool, and `1.10` being just `1.1`. What would you think about using (and installing if it isn't yet available) `jq` for this? Using jq guarantees the output will be correct and that we didn't make an error in indentation, whitespace, and also makes sure that all of the values are properly quoted.

Here is an example of what that would look like:

```bash

TINKERBELL_NETWORK_INTERFACE=10.5.3.1
TINKERBELL_HOST_IP=10.5.3.2
TINKERBELL_NGINX_IP=10.5.3.3
cidr=24

jq -n \
   --arg interface "$TINKERBELL_NETWORK_INTERFACE" \
   --arg cidr "$cidr" \
   --arg host_ip "$TINKERBELL_HOST_IP" \
   --arg nginx_ip "$TINKERBELL_NGINX_IP" \
   '{
  network: "hi",
  renderer: "networkd",
  ethernets: {
    ($interface): {
      addresses: [
        "\($host_ip)/\($cidr)",
        "\($nginx_ip)/\($cidr)"
      ]
    }
  }
}'
```

This prints:

```
{
  "network": "hi",
  "renderer": "networkd",
  "ethernets": {
    "10.5.3.1": {
      "addresses": [
        "10.5.3.2/24",
        "10.5.3.3/24"
      ]
    }
  }
}
```

since JSON is valid YAML, this output works directly with netplan.